### PR TITLE
Send the next WHO only after the previous WHO reply ends

### LINKED ISSUE #238: WHO spam
#232 will introduce code that sends WHO when joining a channel. ThePhilgrim said this can result in excess flood errors. I will likely run into that myself, as I have a lot of channels set on autojoin (see screenshot in README).

### LINKED ISSUE #246: commented out send WHO
In fa9b6368a3ec9da4ff53b55a1ddff1843b76db6d I commented out a `server_view.core.send(f"WHO {channel}")` line from `_handle_join()`. Before commenting it out, I couldn't connect to libera. Even with that line commented out, the away status seems to work fine, because there's a second `server_view.core.send(f"WHO {channel}")` in `_handle_endofnames()`.

We should figure out whether having two WHO sends was intentional, and if not, clean up the mess of commented out code.

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #238. Filtering the output of `--verbose` to only the lines that contain `WHO`, I get this on libera when I connect:

```
Send: b'WHO #c\r\n'
Recv: b':strontium.libera.chat 315 Akuli #c :End of /WHO list.\r\n'
Send: b'WHO #devuan-offtopic\r\n'
Recv: b':strontium.libera.chat 315 Akuli #devuan-offtopic :End of /WHO list.\r\n'
Send: b'WHO #devuan\r\n'
Recv: b':strontium.libera.chat 315 Akuli #devuan :End of /WHO list.\r\n'
Send: b'WHO #ircv3\r\n'
Recv: b':strontium.libera.chat 315 Akuli #ircv3 :End of /WHO list.\r\n'
Send: b'WHO #8banana\r\n'
Recv: b':strontium.libera.chat 315 Akuli #8banana :End of /WHO list.\r\n'
Send: b'WHO ##akuli-testing\r\n'
Recv: b':strontium.libera.chat 315 Akuli ##akuli-testing :End of /WHO list.\r\n'
Send: b'WHO ##guitar\r\n'
Recv: b':strontium.libera.chat 315 Akuli ##guitar :End of /WHO list.\r\n'
Send: b'WHO #tcl\r\n'
Recv: b':strontium.libera.chat 315 Akuli #tcl :End of /WHO list.\r\n'
Send: b'WHO ##programming\r\n'
Recv: b':strontium.libera.chat 315 Akuli ##programming :End of /WHO list.\r\n'
Send: b'WHO #libera\r\n'
Recv: b':strontium.libera.chat 315 Akuli #libera :End of /WHO list.\r\n'
Send: b'WHO #mantaray\r\n'
Recv: b':strontium.libera.chat 315 Akuli #mantaray :End of /WHO list.\r\n'
Send: b'WHO ##learnmath\r\n'
Recv: b':strontium.libera.chat 315 Akuli ##learnmath :End of /WHO list.\r\n'
Send: b'WHO ##learnpython\r\n'
Recv: b':strontium.libera.chat 315 Akuli ##learnpython :End of /WHO list.\r\n'
```

Also fixes #246.

### CODE REVIEW

File: mantaray/received.py [hunk 1]
```diff
@@ -416,6 +413,12 @@ def _handle_namreply(server_view: views.ServerView, args: list[str]) -> None:
     join.nicks.extend(name.lstrip("~&@%+") for name in names.split())
 
 
+# While waiting for a response to a WHO, don't send another WHO.
+# This prevents the server from deciding to disconnect because it's
+# being asked to send a lot of data quickly.
```
Comment (Akuli) [inline mantaray/received.py:418]:
```suggestion
# being asked to send a lot of data quickly.
#
# TODO: clear this when reconnecting
```

**Review by ThePhilgrim — APPROVED** [review]
To the best of my knowledge, this looks good. I'm slightly confused about why there is `.send(WHO ...)` in both `_handle_endofnames` and in `_handle_endofwho`, though. Maybe you can explain?

**Comment by Akuli** [comment]
Responding to WHO is slow. The server sends a lot of information about user, even more information than in the NAMES list.

Previously WHO was sent only in `_handle_endofnames()`. This meant that when joining 10 different channels, mantaray sent WHO for each channel at about the same time. Then the server simultaneously responded to the 10 different WHOs. This happens to work for me with libera, but it probably would cause flood problems with more channels.

Now the logic in `_handle_endofnames()` is a bit different: it ensures the server is never working on multiple WHOs at the same time. If we have sent a WHO but not heard back from it, we don't send anything, we just add the channel name to a list. When the server is eventually done with the first WHO, `_handle_endofwho()` then sends the next WHO, and also removes it from the list.

Do the two sends now make sense?

**Comment by ThePhilgrim** [comment]
@Akuli Yeah, it makes perfect sense. If you don't have anything else to add, I would say this is ready to merge.

**Pull request merged**